### PR TITLE
feat: implement resource metadata annotations

### LINKED ISSUE #135: add 'annotations' to resource metadata
Same as 'labels', but no way to filter on annotations.

### PR DESCRIPTION
**Pull request by smira**
The implementation is exactly same for labels and annotations, but annotations don't support any kind of querying/filtering.

Fixes #135

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Comment by smira** [comment]
See also: https://github.com/cosi-project/specification/pull/4

File: pkg/resource/internal/kv/kv.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+func (kv *KV) Delete(key string) {
+	if _, ok := kv.m[key]; !ok {
+		// no change
+		return
+	}
+
+	kvCopy := make(map[string]string, len(kv.m))
+
+	for k, v := range kv.m {
+		kvCopy[k] = v
```
Comment (frezbo) [inline pkg/resource/internal/kv/kv.go:35]: is it easier to check if the key to delete matches here and just continue?

**Review by frezbo — APPROVED** [review]

File: pkg/resource/internal/kv/kv.go (hunk 1 above)
Comment (smira) [reply pkg/resource/internal/kv/kv.go:39]: fixed!

**Review by frezbo — APPROVED** [review]

**Pull request merged**